google/auth/compute_engine/credentials.py

```diff
@@ -228,22 +247,76 @@ def _make_authorization_grant_assertion(self):
 
         return token
 
-    @_helpers.copy_docstring(credentials.Credentials)
+    def _call_metadata_identity_endpoint(self, request):
+        """Request ID token from metadata identity endpoint.
+
+        Args:
+            request (google.auth.transport.Request): The object used to make
+                HTTP requests.
+
+        Raises:
+            google.auth.exceptions.RefreshError: If the Compute Engine metadata
+                service can't be reached or if the instance has no credentials.
+            ValueError: If extracting expiry from the obtained ID token fails.
+        """
+        try:
+            id_token = _metadata.get(
+                request,
+                "instance/service-accounts/default/identity?audience={}".format(
```

Comment:
There is an issue with the GCE metadata server where the ID Token returned does not include `email` or `email_verified` fields in the ID Token claims.

This is an issue because IAP rejects any ID tokens without an `email` claim. So any requests from GCE to an IAP-protected resource will fail.

The workaround is to include the query parameter `format=full` in the identity endpoint request. I don't like using `format=full` because it returns a LOT of information in the ID token. Also, we expect to have this issue fixed in the coming months, and if we remove `format=full`, that will be a BC breaking change. But this is probably better than releasing a broken integration.

Reply:
added.